File: servNatMan/routers/teams.py
from fastapi import APIRouter, HTTPException, Depends
import psycopg2
from database import get_db_connection
from models import TeamCreate, TeamResponse

router = APIRouter(prefix="/teams", tags=["teams"])

@router.get("/", response_model=dict)
async def get_teams(conn = Depends(get_db_connection)):
    with conn.cursor() as cur:
        cur.execute("SELECT * FROM teams")
        teams = cur.fetchall()
    return {"teams": teams}

@router.post("/", response_model=dict)
async def create_team(team: TeamCreate, conn = Depends(get_db_connection)):
    with conn.cursor() as cur:
        try:
            cur.execute("""
                INSERT INTO teams (team_name) VALUES (%s) RETURNING id
            """, (team.team_name,))
            
            new_team_id = cur.fetchone()['id']
            conn.commit()
            return {"message": "Команда создана", "team_id": new_team_id}
            
        except psycopg2.IntegrityError:
            conn.rollback()
            raise HTTPException(status_code=400, detail="Команда с таким именем уже существует")
        except Exception as e:
            conn.rollback()
            raise HTTPException(status_code=500, detail=f"Ошибка: {str(e)}")

File: NatManApp/src/services/teams.js
import axios from "axios";

const API_URL = "http://217.114.14.77:8002/teams";

// Получить команды
export async function getTeams() {
    try {
        const response = await axios.get(API_URL);
        return response.data.teams;  // список команд
    } catch (error) {
        console.error("Ошибка при получении команд:", error);
        return null;
    }
}

// Создать команду
export async function createTeam(teamName) {
    try {
        const response = await axios.post(API_URL, {
            team_name: teamName
        });
        return response.data;
    } catch (error) {
        console.error("Ошибка при создании команды:", error);
        return null;
    }
}